### PR TITLE
Make method names more selectivity specific

### PR DESCRIPTION
**Pull request by pusewicz**

### CODE REVIEW

File: lib/selectivity/rspec.rb [hunk 1]
```diff
@@ -44,25 +44,25 @@ def find_selectivity_input(from, options)
...
         input.first('.selectivity-multiple-input-container').present?
       end
 
-      def select!(input, item)
+      def _selectivity_select!(input, item)
         input.click
 
         within 'div.selectivity-dropdown' do
-          if (find(:xpath, "//div[@class='selectivity-result-item'][contains(text(), '#{item}')]")).visible?
+          if find('div.selectivity-result-item', text: item).visible?
```
Comment (msx2) [inline lib/selectivity/rspec.rb:55]: This is an improvement only on a first sight. Once you have a lot of options in the select (e.g. https://github.com/msx2/selectivity-rails-sample-app), it takes _a lot_ more time to browse through all the options and find a specific one than using `xpath` ;)

**Comment by msx2** [comment]
@pusewicz Thanks for the changes! One thing about the `xpath` change and I believe this will be ready to merge.

Out of curiosity - you mentioned using an object instead of `_selectivity` prefix. Why such change of heart? :smile:

File: lib/selectivity/rspec.rb (hunk 1 above)
Comment (pusewicz) [inline lib/selectivity/rspec.rb:55]: Give me numbers first. Also, previous version simply did not work.

**Comment by pusewicz** [comment]
I had this version already running in my forked repo :)

File: lib/selectivity/rspec.rb (hunk 1 above)
Comment (msx2) [inline lib/selectivity/rspec.rb:55]: What was the case it didn't work for? What value(s)?

Comment (msx2) [inline lib/selectivity/rspec.rb:55]: As for the numbers, the most primitive metric I came up with:

``` ruby
def select!(input, item)
  started_at = Time.now.strftime('%Y%m%d%H%M%S%L')
  puts "select! #{item}: started at #{started_at}"

  # ... code
  ended_at = Time.now.strftime('%Y%m%d%H%M%S%L')
  puts "select! #{item}: ended at #{ended_at}"
  puts "difference: #{ended_at.to_i - started_at.to_i}"
end
```

Both test cases are from sample application.

### xpath

```
msx2@air-konrad ~/c/o/selectivity-rails-sample-app> bin/rspec spec/features/form_spec.rb
select! Netherlands: started at 20150512182909009
select! Netherlands: ended at 20150512182909978
difference: 969
select! Cyprus: started at 20150512182910423
select! Cyprus: ended at 20150512182911141
difference: 718
select! Russia: started at 20150512182911141
select! Russia: ended at 20150512182911811
difference: 670
select! United Kingdom: started at 20150512182919640
select! United Kingdom: ended at 20150512182920289
difference: 649
select! Spain: started at 20150512182920535
select! Spain: ended at 20150512182921159
difference: 624
select! Germany: started at 20150512182923917
select! Germany: ended at 20150512182924571
difference: 654
select! Poland: started at 20150512182925305
select! Poland: ended at 20150512182925978
difference: 673
.select! Netherlands: started at 20150512182929348
select! Netherlands: ended at 20150512182929985
difference: 637
select! Cyprus: started at 20150512182930242
select! Cyprus: ended at 20150512182930926
difference: 684
select! Russia: started at 20150512182930926
select! Russia: ended at 20150512182931604
difference: 678
select! United Kingdom: started at 20150512182939625
select! United Kingdom: ended at 20150512182940280
difference: 655
select! Spain: started at 20150512182940549
select! Spain: ended at 20150512182941192
difference: 643
select! Germany: started at 20150512182943940
select! Germany: ended at 20150512182944592
difference: 652
select! Poland: started at 20150512182945300
select! Poland: ended at 20150512182945947
difference: 647
.select! Netherlands: started at 20150512182949329
select! Netherlands: ended at 20150512182949992
difference: 663
select! Cyprus: started at 20150512182950274
select! Cyprus: ended at 20150512182950975
difference: 701
select! Russia: started at 20150512182950975
select! Russia: ended at 20150512182951643
difference: 668
select! United Kingdom: started at 20150512182959550
select! United Kingdom: ended at 20150512183000232
difference: 40682
select! Spain: started at 20150512183000508
select! Spain: ended at 20150512183001141
difference: 633
select! Germany: started at 20150512183003886
select! Germany: ended at 20150512183004538
difference: 652
select! Poland: started at 20150512183005271
select! Poland: ended at 20150512183005924
difference: 653
.select! Netherlands: started at 20150512183009309
select! Netherlands: ended at 20150512183009991
difference: 682
select! Cyprus: started at 20150512183010278
select! Cyprus: ended at 20150512183010957
difference: 679
select! Russia: started at 20150512183010957
select! Russia: ended at 20150512183011581
difference: 624
select! United Kingdom: started at 20150512183019569
select! United Kingdom: ended at 20150512183020251
difference: 682
select! Spain: started at 20150512183020516
select! Spain: ended at 20150512183021157
difference: 641
select! Germany: started at 20150512183023888
select! Germany: ended at 20150512183024541
difference: 653
select! Poland: started at 20150512183025260
select! Poland: ended at 20150512183025913
difference: 653
.select! Netherlands: started at 20150512183029268
select! Netherlands: ended at 20150512183029938
difference: 670
select! Cyprus: started at 20150512183030220
select! Cyprus: ended at 20150512183030906
difference: 686
select! Russia: started at 20150512183030907
select! Russia: ended at 20150512183031548
difference: 641
select! United Kingdom: started at 20150512183039701
select! United Kingdom: ended at 20150512183040434
difference: 733
select! Spain: started at 20150512183040703
select! Spain: ended at 20150512183041413
difference: 710
select! Germany: started at 20150512183044160
select! Germany: ended at 20150512183044852
difference: 692
select! Poland: started at 20150512183045601
select! Poland: ended at 20150512183046270
difference: 669
.select! Netherlands: started at 20150512183049756
select! Netherlands: ended at 20150512183050446
difference: 690
select! Cyprus: started at 20150512183050705
select! Cyprus: ended at 20150512183051399
difference: 694
select! Russia: started at 20150512183051399
select! Russia: ended at 20150512183052073
difference: 674
select! United Kingdom: started at 20150512183100077
select! United Kingdom: ended at 20150512183100747
difference: 670
select! Spain: started at 20150512183101017
select! Spain: ended at 20150512183101666
difference: 649
select! Germany: started at 20150512183104381
select! Germany: ended at 20150512183105054
difference: 673
select! Poland: started at 20150512183105799
select! Poland: ended at 20150512183106451
difference: 652
.select! Netherlands: started at 20150512183109863
select! Netherlands: ended at 20150512183110524
difference: 661
select! Cyprus: started at 20150512183110824
select! Cyprus: ended at 20150512183111517
difference: 693
select! Russia: started at 20150512183111517
select! Russia: ended at 20150512183112155
difference: 638
select! United Kingdom: started at 20150512183120172
select! United Kingdom: ended at 20150512183120868
difference: 696
select! Spain: started at 20150512183121126
select! Spain: ended at 20150512183121781
difference: 655
select! Germany: started at 20150512183124511
select! Germany: ended at 20150512183125180
difference: 669
select! Poland: started at 20150512183125916
select! Poland: ended at 20150512183126569
difference: 653
.select! Netherlands: started at 20150512183129934
select! Netherlands: ended at 20150512183130594
difference: 660
select! Cyprus: started at 20150512183130883
select! Cyprus: ended at 20150512183131530
difference: 647
select! Russia: started at 20150512183131530
select! Russia: ended at 20150512183132161
difference: 631
select! United Kingdom: started at 20150512183140268
select! United Kingdom: ended at 20150512183140939
difference: 671
select! Spain: started at 20150512183141197
select! Spain: ended at 20150512183141853
difference: 656
select! Germany: started at 20150512183144617
select! Germany: ended at 20150512183145265
difference: 648
select! Poland: started at 20150512183146036
select! Poland: ended at 20150512183146681
difference: 645
.

Finished in 2 minutes 50.2 seconds (files took 4.16 seconds to load)
8 examples, 0 failures
```

### find

```
msx2@air-konrad ~/c/o/selectivity-rails-sample-app> bin/rspec spec/features/form_spec.rb
select! Netherlands: started at 20150512183722422
select! Netherlands: ended at 20150512183727957
difference: 5535
select! Cyprus: started at 20150512183728257
select! Cyprus: ended at 20150512183732659
difference: 4402
select! Russia: started at 20150512183732659
select! Russia: ended at 20150512183736933
difference: 4274
select! United Kingdom: started at 20150512183744720
select! United Kingdom: ended at 20150512183748711
difference: 3991
select! Spain: started at 20150512183748974
select! Spain: ended at 20150512183752977
difference: 4003
select! Germany: started at 20150512183755680
select! Germany: ended at 20150512183759725
difference: 4045
select! Poland: started at 20150512183800407
select! Poland: ended at 20150512183804347
difference: 3940
.select! Netherlands: started at 20150512183807650
select! Netherlands: ended at 20150512183811768
difference: 4118
select! Cyprus: started at 20150512183812054
select! Cyprus: ended at 20150512183816050
difference: 3996
select! Russia: started at 20150512183816052
select! Russia: ended at 20150512183819914
difference: 3862
select! United Kingdom: started at 20150512183827708
select! United Kingdom: ended at 20150512183831738
difference: 4030
select! Spain: started at 20150512183831992
select! Spain: ended at 20150512183835932
difference: 3940
select! Germany: started at 20150512183838669
select! Germany: ended at 20150512183842753
difference: 4084
select! Poland: started at 20150512183843447
select! Poland: ended at 20150512183847505
difference: 4058
.select! Netherlands: started at 20150512183850800
select! Netherlands: ended at 20150512183854808
difference: 4008
select! Cyprus: started at 20150512183855061
select! Cyprus: ended at 20150512183859022
difference: 3961
select! Russia: started at 20150512183859022
select! Russia: ended at 20150512183902863
difference: 43841
select! United Kingdom: started at 20150512183910664
select! United Kingdom: ended at 20150512183914692
difference: 4028
select! Spain: started at 20150512183914946
select! Spain: ended at 20150512183918904
difference: 3958
select! Germany: started at 20150512183921652
select! Germany: ended at 20150512183925677
difference: 4025
select! Poland: started at 20150512183926384
select! Poland: ended at 20150512183930454
difference: 4070
.select! Netherlands: started at 20150512183933792
select! Netherlands: ended at 20150512183937883
difference: 4091
select! Cyprus: started at 20150512183938147
select! Cyprus: ended at 20150512183942128
difference: 3981
select! Russia: started at 20150512183942128
select! Russia: ended at 20150512183946045
difference: 3917
select! United Kingdom: started at 20150512183953882
select! United Kingdom: ended at 20150512183957886
difference: 4004
select! Spain: started at 20150512183958171
select! Spain: ended at 20150512184002208
difference: 44037
select! Germany: started at 20150512184004970
select! Germany: ended at 20150512184009133
difference: 4163
select! Poland: started at 20150512184009847
select! Poland: ended at 20150512184013923
difference: 4076
.select! Netherlands: started at 20150512184017349
select! Netherlands: ended at 20150512184021493
difference: 4144
select! Cyprus: started at 20150512184021774
select! Cyprus: ended at 20150512184025753
difference: 3979
select! Russia: started at 20150512184025753
select! Russia: ended at 20150512184029704
difference: 3951
select! United Kingdom: started at 20150512184037680
select! United Kingdom: ended at 20150512184041808
difference: 4128
select! Spain: started at 20150512184042127
select! Spain: ended at 20150512184046228
difference: 4101
select! Germany: started at 20150512184048883
select! Germany: ended at 20150512184053062
difference: 4179
select! Poland: started at 20150512184053782
select! Poland: ended at 20150512184057905
difference: 4123
.select! Netherlands: started at 20150512184101314
select! Netherlands: ended at 20150512184105414
difference: 4100
select! Cyprus: started at 20150512184105701
select! Cyprus: ended at 20150512184109778
difference: 4077
select! Russia: started at 20150512184109778
select! Russia: ended at 20150512184113831
difference: 4053
select! United Kingdom: started at 20150512184121618
select! United Kingdom: ended at 20150512184125773
difference: 4155
select! Spain: started at 20150512184126094
select! Spain: ended at 20150512184130196
difference: 4102
select! Germany: started at 20150512184132876
select! Germany: ended at 20150512184137045
difference: 4169
select! Poland: started at 20150512184137779
select! Poland: ended at 20150512184141980
difference: 4201
.select! Netherlands: started at 20150512184145321
select! Netherlands: ended at 20150512184149536
difference: 4215
select! Cyprus: started at 20150512184149813
select! Cyprus: ended at 20150512184153946
difference: 4133
select! Russia: started at 20150512184153946
select! Russia: ended at 20150512184158001
difference: 4055
select! United Kingdom: started at 20150512184205832
select! United Kingdom: ended at 20150512184210017
difference: 4185
select! Spain: started at 20150512184210328
select! Spain: ended at 20150512184214472
difference: 4144
select! Germany: started at 20150512184217186
select! Germany: ended at 20150512184221403
difference: 4217
select! Poland: started at 20150512184222130
select! Poland: ended at 20150512184226243
difference: 4113
.select! Netherlands: started at 20150512184229666
select! Netherlands: ended at 20150512184233859
difference: 4193
select! Cyprus: started at 20150512184234127
select! Cyprus: ended at 20150512184238263
difference: 4136
select! Russia: started at 20150512184238263
select! Russia: ended at 20150512184242233
difference: 3970
select! United Kingdom: started at 20150512184250110
select! United Kingdom: ended at 20150512184254285
difference: 4175
select! Spain: started at 20150512184254558
select! Spain: ended at 20150512184258652
difference: 4094
select! Germany: started at 20150512184301376
select! Germany: ended at 20150512184305604
difference: 4228
select! Poland: started at 20150512184306484
select! Poland: ended at 20150512184310725
difference: 4241
.

Finished in 6 minutes 1 second (files took 3.26 seconds to load)
8 examples, 0 failures
```

Comment (pusewicz) [inline lib/selectivity/rspec.rb:55]: It does not work for this:

image

I'm using the Poltergeist/PhantomJS driver. Using it with:

``` ruby
selectivity_select 'Mustang', from: '#car_offer_model_slug'
```

Comment (msx2) [inline lib/selectivity/rspec.rb:55]: Alright, thanks for details! I think it has already been reported - https://github.com/teampoltergeist/poltergeist/issues/317.

I admit I didn't check on Poltergeist though. Will merge this now, play with the driver and try to find out a sufficient solution that uses `xpath`. You are also very welcome to try yourself :wink:

**Pull request merged**